packages/hooks/src/hooks.spec.js

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect } from 'react';
+import { act, renderHook } from 'react-hooks-testing-library';
+import { render, cleanup } from 'react-testing-library';
+import {
+  useDocumentEventListener,
+  useElementNode,
+  useViewportSize,
+} from './index';
+
+afterAll(cleanup);
+
+describe('packages/hooks', () => {
+  describe('useDocumentEventListener', () => {
+    test('event callback should fire when enabled is true', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() => useDocumentEventListener('click', eventCallback));
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(1);
+    });
+
+    test('event callback should not fire when enabled is false', () => {
+      const eventCallback = jest.fn();
+
+      renderHook(() =>
+        useDocumentEventListener({
+          type: 'click',
+          eventCallback,
+          enabled: false,
+        }),
+      );
+
+      act(() => {
+        document.body.dispatchEvent(new MouseEvent('click', { bubbles: true }));
+      });
+
+      expect(eventCallback).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe.skip('useMutationObserver', () => {}); //eslint-disable-line jest/no-disabled-tests
+
+  describe('useElementNode', () => {
+    let count = 1;
```

Comment:
Not sure it matters here, but could use `useState` instead...

Reply:
useEffect to make sure it's capturing the lifecycle update